### DISCUSSION TITLE
How to enable DLNA functionality in stashapp/stash deployed in Synology's Container Manager?

### DISCUSSION THREAD

**Discussion by onlyhooops** [stashapp/stash#4495] [Q&A] [answered]
On my Synology DS923+ NAS, I deployed the latest stashapp/stash using Container Manager. It runs well, but the DLNA feature cannot be effectively enabled. I have mapped the container port (9999) to port 10001 on the physical machine. After enabling the DLNA feature, I am unable to find any server information on any device. Please advise on how to proceed. Thank you very much for your help.

**Comment by JanJastrow** [comment, marked as answer]
DLNA uses additional/different ports: 8200 & 1900 and these are in probably in use by the Synology.
You need to disable all internal DLNA server of the Synology NAS (like `Video Station`, etc.)

Yes, you can Map 9999 to a different port, but thats only the Web interface.
I had issues forwarding 8200 & 1900 through Docker. (Syn complained)

In my experience no external player (TVs, KODI, etc) are capable of discovering DLNA on any other port than the ones mentioned above, so that's also no solution.

My solution in the past was to connect a second ethernet cable to my NAS (which had 2 ports) and run the container on the "host" network. That was the only way I could make it work.

**Reply by onlyhooops** [reply]
Thank you for your patient and detailed response.

However, following your instructions, I manually set the ports to 8200 & 1900 and restarted stashapp/stash. After reopening the DLNA service, I still cannot locate the DLNA server on my local network. I can assure you that my Synology device is not using ports 8200 & 1900. 

Regarding what you mentioned about "My solution in the past was to connect a second ethernet cable to my NAS (which had 2 ports) and run the container on the 'host' network. That was the only way I could make it work," I'm not quite sure I understand. I have now connected an additional cable to my Synology device, but I've found that I cannot directly change the network settings for the deployed container in Container Manager. 

If you have the time, I would greatly appreciate a more detailed explanation.

P.S.: DLNA service is crucial for me.

**Reply by JanJastrow** [reply]
Hi again,
as far as I know you can't change an already created docker container to the `Host` network.
This has to be done while creating it:
![grafik](https://github.com/stashapp/stash/assets/2591991/a45ae382-ca18-4f1f-995f-c43a06fd3197)

So create a new container with the same settings and run it.
Stash should be available at `http://your-second-synology-lan-ip:9999` (See Syno > Control Panel > Network > Network Interfaces).
And now (at least in my case) I was also able to find my installation with KODI or other DLNA clients.

**Reply by onlyhooops** [reply]
Success! Following your guidance, I successfully redeployed the container on the host network, and the DLNA server is now up and running! Thank you for your patient assistance. Wishing you all the best.
